R/check_pkg_installed.R

```diff
@@ -44,7 +44,9 @@ NULL
 #' @export
 check_pkg_installed <- function(pkg,
                                 reference_pkg = "cards",
-                                call = parent.frame()) {
+                                call = get_cli_abort_call()) {
```

Comment:
I think this is a case where we may want to keep the call argument because this function is meant to mimic the rlang version, which also has the call arg.

Also, since this is an exported function, we can't make its default value an unexported function.

Maybe we don't need to export these check pkg functions, and we can just add them to their own separate standalone script? I am leaning towards this....

Reply:
We could also leave the default `call = parent.frame()`. But each time we call this function, we'd need to specify `call = get_cli_abort_call()`.

Anyway, let's discuss!

Reply:
I think it works well if we leave the `call = get_cli_abort_call()` argument, but we can discuss!

Reply:
let's create three different standalone files
1. checks (we already have, but will need to update it to use the set and get cli env functions)
2. pkg install check functions
3. the standalone that defines the set and get env functions.

The first two will depend on the third, and we specify this dependency with `dependencies: standalone-obj-type.R` in the YAML header.

![image](https://github.com/insightsengineering/cards/assets/26774684/d0176e14-da1e-4e6a-939e-298d467ecd1e)